[v2 branch] The way of exporting module seems not friendly in package.json

### Description

In my project, I use typescript(v4.7) with config `"module": "esnext", "moduleResolution": "node"`.

If I use the lasted release of the tauri plugin package(eg: @tauri-apps/plugin-log). I will get type error which says can't recognize package `@tauri-apps/plugin-log`. Then I check the package.json of the plugin, I notice below code.

https://github.com/tauri-apps/plugins-workspace/blob/5a8bbe296704f679063ef8b820d582d7f728400b/plugins/log/package.json#L9-L14

After some research, they told me should use typescript(v5) with config `"module": "esnext", "moduleResolution": "bundler"`. They are right. The plugin package now are recognizable.

But new type error appears. `tsc` now can't recognize subpath of package. Eg:

```ts
// Cannot find module '@zing/helpers/dist/tippy' or its corresponding type declarations.
import { getTippy } from '@zing/helpers/dist/tippy';

// package.json of @zing/helpers
{
  ...
  "main": "dist/index.js",
  "module": "dist/index.js",
  "types": "dist/index.d.ts",
  "files": ["dist"],
  "exports": {
    ".": "./dist/index.js",
    "./dist/": "./dist/"
  },
}
```

### Hope

Change the way of exporting module to a more friendly one. 

Here is a example. https://github.com/ProseMirror/prosemirror-model/blob/master/package.json#L5-L12

![](https://bashupload.com/dQ0hb/SMWGb.png)

Well, I switch back to typescript(v4.7) with config `"module": "esnext", "moduleResolution": "node"`.

```ts
// get type error, but the app can run well
import log from '@tauri-apps/plugin-log';

// no type error, but get a vite runtime error
import log from '@tauri-apps/plugin-log/dist-js/index';

// runtime error
✘ [ERROR] Missing "./dist-js/index" specifier in "@tauri-apps/plugin-log" package [plugin vite:dep-scan]

    src/services/tauri.ts:9:21:
      9 │ export * as log from '@tauri-apps/plugin-log/dist-js/index';
        ╵                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  This error came from the "onResolve" callback registered here:

    node_modules/.pnpm/esbuild@0.18.20/node_modules/esbuild/lib/main.js:1292:20:
      1292 │       let promise = setup({
```

--- a/plugins/authenticator/package.json
+++ b/plugins/authenticator/package.json
@@ -7,6 +7,9 @@
     "Tauri Programme within The Commons Conservancy"
   ],
   "type": "module",
+  "types": "./dist-js/index.d.ts",
+  "main": "./dist-js/index.cjs",
+  "module": "./dist-js/index.js",
   "exports": {
     "types": "./dist-js/index.d.ts",
     "import": "./dist-js/index.js",

--- a/plugins/autostart/package.json
+++ b/plugins/autostart/package.json
@@ -6,6 +6,9 @@
     "Tauri Programme within The Commons Conservancy"
   ],
   "type": "module",
+  "types": "./dist-js/index.d.ts",
+  "main": "./dist-js/index.cjs",
+  "module": "./dist-js/index.js",
   "exports": {
     "types": "./dist-js/index.d.ts",
     "import": "./dist-js/index.js",

--- a/plugins/barcode-scanner/package.json
+++ b/plugins/barcode-scanner/package.json
@@ -7,6 +7,9 @@
     "Tauri Programme within The Commons Conservancy"
   ],
   "type": "module",
+  "types": "./dist-js/index.d.ts",
+  "main": "./dist-js/index.cjs",
+  "module": "./dist-js/index.js",
   "exports": {
     "types": "./dist-js/index.d.ts",
     "import": "./dist-js/index.js",

--- a/plugins/cli/package.json
+++ b/plugins/cli/package.json
@@ -6,6 +6,9 @@
     "Tauri Programme within The Commons Conservancy"
   ],
   "type": "module",
+  "types": "./dist-js/index.d.ts",
+  "main": "./dist-js/index.cjs",
+  "module": "./dist-js/index.js",
   "exports": {
     "types": "./dist-js/index.d.ts",
     "import": "./dist-js/index.js",

--- a/plugins/clipboard-manager/package.json
+++ b/plugins/clipboard-manager/package.json
@@ -6,6 +6,9 @@
     "Tauri Programme within The Commons Conservancy"
   ],
   "type": "module",
+  "types": "./dist-js/index.d.ts",
+  "main": "./dist-js/index.cjs",
+  "module": "./dist-js/index.js",
   "exports": {
     "types": "./dist-js/index.d.ts",
     "import": "./dist-js/index.js",

--- a/plugins/deep-link/package.json
+++ b/plugins/deep-link/package.json
@@ -7,6 +7,9 @@
     "Tauri Programme within The Commons Conservancy"
   ],
   "type": "module",
+  "types": "./dist-js/index.d.ts",
+  "main": "./dist-js/index.cjs",
+  "module": "./dist-js/index.js",
   "exports": {
     "types": "./dist-js/index.d.ts",
     "import": "./dist-js/index.js",

--- a/plugins/dialog/package.json
+++ b/plugins/dialog/package.json
@@ -6,6 +6,9 @@
     "Tauri Programme within The Commons Conservancy"
   ],
   "type": "module",
+  "types": "./dist-js/index.d.ts",
+  "main": "./dist-js/index.cjs",
+  "module": "./dist-js/index.js",
   "exports": {
     "types": "./dist-js/index.d.ts",
     "import": "./dist-js/index.js",

--- a/plugins/fs/package.json
+++ b/plugins/fs/package.json
@@ -7,6 +7,9 @@
     "Tauri Programme within The Commons Conservancy"
   ],
   "type": "module",
+  "types": "./dist-js/index.d.ts",
+  "main": "./dist-js/index.cjs",
+  "module": "./dist-js/index.js",
   "exports": {
     "types": "./dist-js/index.d.ts",
     "import": "./dist-js/index.js",

--- a/plugins/global-shortcut/package.json
+++ b/plugins/global-shortcut/package.json
@@ -6,6 +6,9 @@
     "Tauri Programme within The Commons Conservancy"
   ],
   "type": "module",
+  "types": "./dist-js/index.d.ts",
+  "main": "./dist-js/index.cjs",
+  "module": "./dist-js/index.js",
   "exports": {
     "types": "./dist-js/index.d.ts",
     "import": "./dist-js/index.js",

--- a/plugins/http/package.json
+++ b/plugins/http/package.json
@@ -6,6 +6,9 @@
     "Tauri Programme within The Commons Conservancy"
   ],
   "type": "module",
+  "types": "./dist-js/index.d.ts",
+  "main": "./dist-js/index.cjs",
+  "module": "./dist-js/index.js",
   "exports": {
     "types": "./dist-js/index.d.ts",
     "import": "./dist-js/index.js",

--- a/plugins/log/package.json
+++ b/plugins/log/package.json
@@ -7,6 +7,9 @@
     "Tauri Programme within The Commons Conservancy"
   ],
   "type": "module",
+  "types": "./dist-js/index.d.ts",
+  "main": "./dist-js/index.cjs",
+  "module": "./dist-js/index.js",
   "exports": {
     "types": "./dist-js/index.d.ts",
     "import": "./dist-js/index.js",

--- a/plugins/notification/package.json
+++ b/plugins/notification/package.json
@@ -6,6 +6,9 @@
     "Tauri Programme within The Commons Conservancy"
   ],
   "type": "module",
+  "types": "./dist-js/index.d.ts",
+  "main": "./dist-js/index.cjs",
+  "module": "./dist-js/index.js",
   "exports": {
     "types": "./dist-js/index.d.ts",
     "import": "./dist-js/index.js",

--- a/plugins/os/package.json
+++ b/plugins/os/package.json
@@ -6,6 +6,9 @@
     "Tauri Programme within The Commons Conservancy"
   ],
   "type": "module",
+  "types": "./dist-js/index.d.ts",
+  "main": "./dist-js/index.cjs",
+  "module": "./dist-js/index.js",
   "exports": {
     "types": "./dist-js/index.d.ts",
     "import": "./dist-js/index.js",

--- a/plugins/positioner/package.json
+++ b/plugins/positioner/package.json
@@ -7,6 +7,9 @@
     "Tauri Programme within The Commons Conservancy"
   ],
   "type": "module",
+  "types": "./dist-js/index.d.ts",
+  "main": "./dist-js/index.cjs",
+  "module": "./dist-js/index.js",
   "exports": {
     "types": "./dist-js/index.d.ts",
     "import": "./dist-js/index.js",

--- a/plugins/process/package.json
+++ b/plugins/process/package.json
@@ -6,6 +6,9 @@
     "Tauri Programme within The Commons Conservancy"
   ],
   "type": "module",
+  "types": "./dist-js/index.d.ts",
+  "main": "./dist-js/index.cjs",
+  "module": "./dist-js/index.js",
   "exports": {
     "types": "./dist-js/index.d.ts",
     "import": "./dist-js/index.js",

--- a/plugins/shell/package.json
+++ b/plugins/shell/package.json
@@ -6,6 +6,9 @@
     "Tauri Programme within The Commons Conservancy"
   ],
   "type": "module",
+  "types": "./dist-js/index.d.ts",
+  "main": "./dist-js/index.cjs",
+  "module": "./dist-js/index.js",
   "exports": {
     "types": "./dist-js/index.d.ts",
     "import": "./dist-js/index.js",

--- a/plugins/sql/package.json
+++ b/plugins/sql/package.json
@@ -7,6 +7,9 @@
     "Tauri Programme within The Commons Conservancy"
   ],
   "type": "module",
+  "types": "./dist-js/index.d.ts",
+  "main": "./dist-js/index.cjs",
+  "module": "./dist-js/index.js",
   "exports": {
     "types": "./dist-js/index.d.ts",
     "import": "./dist-js/index.js",

--- a/plugins/store/package.json
+++ b/plugins/store/package.json
@@ -7,6 +7,9 @@
     "Tauri Programme within The Commons Conservancy"
   ],
   "type": "module",
+  "types": "./dist-js/index.d.ts",
+  "main": "./dist-js/index.cjs",
+  "module": "./dist-js/index.js",
   "exports": {
     "types": "./dist-js/index.d.ts",
     "import": "./dist-js/index.js",

--- a/plugins/stronghold/package.json
+++ b/plugins/stronghold/package.json
@@ -7,6 +7,9 @@
     "Tauri Programme within The Commons Conservancy"
   ],
   "type": "module",
+  "types": "./dist-js/index.d.ts",
+  "main": "./dist-js/index.cjs",
+  "module": "./dist-js/index.js",
   "exports": {
     "types": "./dist-js/index.d.ts",
     "import": "./dist-js/index.js",

--- a/plugins/updater/package.json
+++ b/plugins/updater/package.json
@@ -6,6 +6,9 @@
     "Tauri Programme within The Commons Conservancy"
   ],
   "type": "module",
+  "types": "./dist-js/index.d.ts",
+  "main": "./dist-js/index.cjs",
+  "module": "./dist-js/index.js",
   "exports": {
     "types": "./dist-js/index.d.ts",
     "import": "./dist-js/index.js",

--- a/plugins/upload/package.json
+++ b/plugins/upload/package.json
@@ -7,6 +7,9 @@
     "Tauri Programme within The Commons Conservancy"
   ],
   "type": "module",
+  "types": "./dist-js/index.d.ts",
+  "main": "./dist-js/index.cjs",
+  "module": "./dist-js/index.js",
   "exports": {
     "types": "./dist-js/index.d.ts",
     "import": "./dist-js/index.js",

--- a/plugins/websocket/package.json
+++ b/plugins/websocket/package.json
@@ -6,6 +6,9 @@
     "Tauri Programme within The Commons Conservancy"
   ],
   "type": "module",
+  "types": "./dist-js/index.d.ts",
+  "main": "./dist-js/index.cjs",
+  "module": "./dist-js/index.js",
   "exports": {
     "types": "./dist-js/index.d.ts",
     "import": "./dist-js/index.js",

--- a/plugins/window-state/package.json
+++ b/plugins/window-state/package.json
@@ -7,6 +7,9 @@
     "Tauri Programme within The Commons Conservancy"
   ],
   "type": "module",
+  "types": "./dist-js/index.d.ts",
+  "main": "./dist-js/index.cjs",
+  "module": "./dist-js/index.js",
   "exports": {
     "types": "./dist-js/index.d.ts",
     "import": "./dist-js/index.js",

--- a/shared/template/package.json
+++ b/shared/template/package.json
@@ -6,6 +6,9 @@
     "Tauri Programme within The Commons Conservancy"
   ],
   "type": "module",
+  "types": "./dist-js/index.d.ts",
+  "main": "./dist-js/index.cjs",
+  "module": "./dist-js/index.js",
   "exports": {
     "types": "./dist-js/index.d.ts",
     "import": "./dist-js/index.js",

